features/FEATURE_LWIP/lwip-interface/lwip_stack.c

```diff
@@ -81,9 +97,24 @@ static struct lwip_socket *mbed_lwip_arena_alloc(void)
     return 0;
 }
 
+static nsapi_error_t mbed_lwip_setsockopt(nsapi_stack_t *stack, nsapi_socket_t handle, int level, int optname, const void *optval, unsigned optlen);
+
 static void mbed_lwip_arena_dealloc(struct lwip_socket *s)
 {
     s->in_use = false;
+
+    while(s->multicast_memberships_count > 0) {
+        static uint32_t index = 0;
+        next_registered_multicast_member(s->multicast_memberships_registery, index);
+        nsapi_ip_mreq_t address = {
+            .imr_multiaddr = s->multicast_memberships[index].imr_multiaddr,
+            .imr_interface = s->multicast_memberships[index].imr_interface
+        };
+        mbed_lwip_setsockopt(0, s, 0, NSAPI_DROP_MEMBERSHIP, &address, sizeof(address));
```

Comment:
You could just pass `&s->multicast_memberships[index]`, no need for the temporary.